platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java

```diff
@@ -152,15 +152,21 @@ private void resetSocketTimeout() throws ReconnectException {
     @NonNull
     public ReservedSignedState execute(@NonNull final SignedStateValidator validator) throws ReconnectException {
         increaseSocketTimeout();
+        ReservedSignedState reservedSignedState = null;
         try {
             receiveSignatures();
-            final ReservedSignedState reservedSignedState = reconnect();
+            reservedSignedState = reconnect();
             validator.validate(reservedSignedState.get(), addressBook, stateValidationData);
             ReconnectUtils.endReconnectHandshake(connection);
             return reservedSignedState;
         } catch (final IOException | SignedStateInvalidException e) {
+            if (reservedSignedState != null) {
+                // if the state was received, we need to release it or it will be leaked
+                reservedSignedState.close();
+            }
             throw new ReconnectException(e);
         } catch (final InterruptedException e) {
+            // an interrupt can only occur in the reconnect() method, so we don't need to close the reservedSignedState
```

Comment:
Minor: you never know what the code above is going to be doing in the future. Since the closing is guarded by a `!= null` check anyway, I really don't see harm in closing the state here as well.

Reply:
Good point, but since this is a cherry pick, probably better to make this change in develop rather than in this particular PR.

Reply:
Agree. Please feel free to resolve the conversation then (preferably, after filing a ticket to implement the suggested improvement.)